lib/krikri/mapper.rb

```diff
@@ -90,23 +90,55 @@ def mapping_exception_desc(rec)
     ##
     # A SoftwareAgent to run mapping processes.
     #
+    # @example
+    #
+    #   To map the records harvested by the harvest activity with ID 1:
+    #
+    #   Krikri::Mapper::Agent.enqueue(
+    #     :mapping,
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       generator_uri: 'http://ldp.local.dp.la/ldp/activity/1'
+    #     }
+    #   )
+    #   Or, supposing you already have a harvest Activity instance named
+    #   `my_activity':
+    #   [ ... ]
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       activity: my_activity
+    #     }
+    #
     # @see: Krikri::SoftwareAgent, Krikri::Activity
     class Agent
       include SoftwareAgent
 
-      attr_reader :name, :generator_uri
+      attr_reader :name
 
       def self.queue_name
         :mapping
       end
 
+      def self.entity_behavior
+        Krikri::AggregationEntityBehavior
+      end
+
+      def self.generator_entity_behavior
```

Comment:
I'm realizing now that this is the reason I had it be a class method:
https://github.com/dpla/KriKri/blob/feature/index-an-activity/app/models/krikri/activity.rb#L101

`agent` is a string field of `Activity`, from the database record that's saved when the job is run.  I need to be able to determine the behavior associated with that agent's class.  I'll ponder alternative ways of doing it, but it's not obvious to me yet how to do it without it being a class method.